### PR TITLE
Added note on --allowed-origins flag

### PR DESCRIPTION
**Pull request by royendo**
Added under get-started where we mention start a new rill project as a note,

added the flag to references/cli/start.md

Based on Issue: https://github.com/rilldata/rill/pull/5365
and Slack: https://rilldata.slack.com/archives/C01UFNH4W3B/p1723481208803969?thread_ts=1723461905.169519&cid=C01UFNH4W3B

### CODE REVIEW

File: docs/docs/home/get-started.md [hunk 1]
```diff
@@ -38,6 +38,11 @@ rill start my-rill-project
 
 The Rill web app runs locally at `http://localhost:9009` and will create code files in the `my-rill-project` directory.
 
+:::note
+Starting from v0.48, we have added the ability to self host the local Rill Developer application on a non-localhost endpoint. Please refer to the [rill start reference page](https://docs.rilldata.com/reference/cli/start) for more details on using the `--allowed-origins` flag. This method is not officially supported, please proceed at your own risk.
+:::
```
Comment (begelundmuller) [inline docs/docs/home/get-started.md:43]: I think we should take this out – we have *not* added the ability to self-host, it is still very much discouraged 😅  The `--allowed-origins` flag is just to provide an escape hatch for the people who do it despite the poor "support" for it.

**Comment by begelundmuller** [comment]
I have pushed a commit that improves the CLI help note for `--allowed-origins`, and regenerated the CLI docs. Seems like regenerating the docs added a few unrelated changes that had been missed earlier.

File: docs/docs/home/get-started.md (hunk 1 above)
Comment (AndrewRTsao) [reply docs/docs/home/get-started.md:43]: We can say that starting in v0.48, we've made changes to the CORS policy to only allow calls from the default HTTP URL by default for security policies. Therefore, if you are hosting Rill Developer (not officially supported), this means that you will need to pass in additional CORS origins via the `--allowed-origins` flag when starting Rill.

Comment (royendo) [reply docs/docs/home/get-started.md]: made the above changes!

**Review by AndrewRTsao — APPROVED** [review]

**Pull request merged**